[Bug] [TDengineSinkWriter] get java.lang.NullPointerException if there are empty value in some column or tag in tdengine source tables

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

 if there are empty value in some column or tag in tdengine source tables, the tdengine sink job got failure and get null exception.
below is exception stack:

java.lang.RuntimeException: java.lang.NullPointerException at org.apache.seatunnel.engine.server.task.flow.SinkFlowLifeCycle.received(SinkFlowLifeCycle.java:302) at org.apache.seatunnel.engine.server.task.flow.SinkFlowLifeCycle.received(SinkFlowLifeCycle.java:70) at org.apache.seatunnel.engine.server.task.SeaTunnelTransformCollector.collect(SeaTunnelTransformCollector.java:39) at org.apache.seatunnel.engine.server.task.SeaTunnelTransformCollector.collect(SeaTunnelTransformCollector.java:27) at org.apache.seatunnel.engine.server.task.group.queue.IntermediateBlockingQueue.handleRecord(IntermediateBlockingQueue.java:75) at org.apache.seatunnel.engine.server.task.group.queue.IntermediateBlockingQueue.collect(IntermediateBlockingQueue.java:50) at org.apache.seatunnel.engine.server.task.flow.IntermediateQueueFlowLifeCycle.collect(IntermediateQueueFlowLifeCycle.java:51) at org.apache.seatunnel.engine.server.task.TransformSeaTunnelTask.collect(TransformSeaTunnelTask.java:72) at org.apache.seatunnel.engine.server.task.SeaTunnelTask.stateProcess(SeaTunnelTask.java:169) at org.apache.seatunnel.engine.server.task.TransformSeaTunnelTask.call(TransformSeaTunnelTask.java:77) at org.apache.seatunnel.engine.server.TaskExecutionService$BlockingWorker.run(TaskExecutionService.java:694) at org.apache.seatunnel.engine.server.TaskExecutionService$NamedTaskWrapper.run(TaskExecutionService.java:1019) at org.apache.seatunnel.api.tracing.MDCRunnable.run(MDCRunnable.java:43) at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) at java.util.concurrent.FutureTask.run(FutureTask.java:266) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) at java.lang.Thread.run(Thread.java:750) Caused by: java.lang.NullPointerException at org.apache.seatunnel.connectors.seatunnel.tdengine.sink.TDengineSinkWriter.lambda$convertDataType$0(TDengineSinkWriter.java:136) at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948) at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:546) at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:505) at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:511) at org.apache.seatunnel.connectors.seatunnel.tdengine.sink.TDengineSinkWriter.convertDataType(TDengineSinkWriter.java:150) at org.apache.seatunnel.connectors.seatunnel.tdengine.sink.TDengineSinkWriter.write(TDengineSinkWriter.java:94) at org.apache.seatunnel.connectors.seatunnel.tdengine.sink.TDengineSinkWriter.write(TDengineSinkWriter.java:51) at org.apache.seatunnel.engine.server.task.flow.SinkFlowLifeCycle.received(SinkFlowLifeCycle.java:268) ... 17 more

### SeaTunnel Version

2.3.9

### SeaTunnel Config

```conf
env {
  # You can set engine configuration here
  parallelism = 1
  job.mode = "BATCH"
  checkpoint.interval = 5000
  read_limit.bytes_per_second=7000000
  read_limit.rows_per_second=500  
}

# source config
source {
  TDengine {
    url : "jdbc:TAOS-RS://tdengine-cluster:6041/"
    username : "root"
    password : "taosdata"
    database : "mmbybuduflmz_devices"
    stable : "iotpro_3b523e67fc7e3dd1"
    lower_bound : "2025-04-01 10:30:56.0"
    upper_bound : "2025-04-02 03:00:00.025"
    plugin_output = "mmbybuduflmz_iotpro_3b523e67fc7e3dd1"
}


}

# sink config
sink {
  TDengine {
    plugin_input = ["mmbybuduflmz_iotpro_3b523e67fc7e3dd1"]
    url : "jdbc:TAOS-RS://10.207.11.1:6041/"
    username : "root"
    password : "taosdata"
    database : "mmbybuduflmz_devices"
    stable : "iotpro_3b523e67fc7e3dd1"
    timezone: UTC
}


}
```

### Running Command

```shell
use rest api to submit a job
```

### Error Exception

```log
java.lang.RuntimeException: java.lang.NullPointerException at org.apache.seatunnel.engine.server.task.flow.SinkFlowLifeCycle.received(SinkFlowLifeCycle.java:302) at org.apache.seatunnel.engine.server.task.flow.SinkFlowLifeCycle.received(SinkFlowLifeCycle.java:70) at org.apache.seatunnel.engine.server.task.SeaTunnelTransformCollector.collect(SeaTunnelTransformCollector.java:39) at org.apache.seatunnel.engine.server.task.SeaTunnelTransformCollector.collect(SeaTunnelTransformCollector.java:27) at org.apache.seatunnel.engine.server.task.group.queue.IntermediateBlockingQueue.handleRecord(IntermediateBlockingQueue.java:75) at org.apache.seatunnel.engine.server.task.group.queue.IntermediateBlockingQueue.collect(IntermediateBlockingQueue.java:50) at org.apache.seatunnel.engine.server.task.flow.IntermediateQueueFlowLifeCycle.collect(IntermediateQueueFlowLifeCycle.java:51) at org.apache.seatunnel.engine.server.task.TransformSeaTunnelTask.collect(TransformSeaTunnelTask.java:72) at org.apache.seatunnel.engine.server.task.SeaTunnelTask.stateProcess(SeaTunnelTask.java:169) at org.apache.seatunnel.engine.server.task.TransformSeaTunnelTask.call(TransformSeaTunnelTask.java:77) at org.apache.seatunnel.engine.server.TaskExecutionService$BlockingWorker.run(TaskExecutionService.java:694) at org.apache.seatunnel.engine.server.TaskExecutionService$NamedTaskWrapper.run(TaskExecutionService.java:1019) at org.apache.seatunnel.api.tracing.MDCRunnable.run(MDCRunnable.java:43) at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) at java.util.concurrent.FutureTask.run(FutureTask.java:266) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) at java.lang.Thread.run(Thread.java:750) Caused by: java.lang.NullPointerException at org.apache.seatunnel.connectors.seatunnel.tdengine.sink.TDengineSinkWriter.lambda$convertDataType$0(TDengineSinkWriter.java:136) at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948) at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:546) at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:505) at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:511) at org.apache.seatunnel.connectors.seatunnel.tdengine.sink.TDengineSinkWriter.convertDataType(TDengineSinkWriter.java:150) at org.apache.seatunnel.connectors.seatunnel.tdengine.sink.TDengineSinkWriter.write(TDengineSinkWriter.java:94) at org.apache.seatunnel.connectors.seatunnel.tdengine.sink.TDengineSinkWriter.write(TDengineSinkWriter.java:51) at org.apache.seatunnel.engine.server.task.flow.SinkFlowLifeCycle.received(SinkFlowLifeCycle.java:268) ... 17 more
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors-v2/connector-tdengine/src/main/java/org/apache/seatunnel/connectors/seatunnel/tdengine/sink/TDengineSinkWriter.java
+++ b/seatunnel-connectors-v2/connector-tdengine/src/main/java/org/apache/seatunnel/connectors/seatunnel/tdengine/sink/TDengineSinkWriter.java
@@ -17,6 +17,7 @@
 
 package org.apache.seatunnel.connectors.seatunnel.tdengine.sink;
 
+import org.apache.seatunnel.shade.com.google.common.annotations.VisibleForTesting;
 import org.apache.seatunnel.shade.com.google.common.base.Throwables;
 import org.apache.seatunnel.shade.com.google.common.collect.Lists;
 import org.apache.seatunnel.shade.com.typesafe.config.Config;
@@ -129,7 +130,8 @@ public class TDengineSinkWriter extends AbstractSinkWriter<SeaTunnelRow, Void> {
         }
     }
 
-    private Object[] convertDataType(Object[] objects) {
+    @VisibleForTesting
+    Object[] convertDataType(Object[] objects) {
         return Arrays.stream(objects)
                 .map(
                         object -> {

--- a/seatunnel-connectors-v2/connector-tdengine/src/main/java/org/apache/seatunnel/connectors/seatunnel/tdengine/sink/TDengineSinkWriter.java
+++ b/seatunnel-connectors-v2/connector-tdengine/src/main/java/org/apache/seatunnel/connectors/seatunnel/tdengine/sink/TDengineSinkWriter.java
@@ -133,6 +133,9 @@ public class TDengineSinkWriter extends AbstractSinkWriter<SeaTunnelRow, Void> {
         return Arrays.stream(objects)
                 .map(
                         object -> {
+                            if (object == null) {
+                                return null;
+                            }
                             if (LocalDateTime.class.equals(object.getClass())) {
                                 // transform timezone according to the config
                                 return "'"

--- a/seatunnel-connectors-v2/connector-tdengine/src/test/java/org/apache/seatunnel/connectors/seatunnel/tdengine/sink/TDengineSinkWriterTest.java
+++ b/seatunnel-connectors-v2/connector-tdengine/src/test/java/org/apache/seatunnel/connectors/seatunnel/tdengine/sink/TDengineSinkWriterTest.java
@@ -29,7 +29,6 @@ import org.apache.seatunnel.shade.com.typesafe.config.ConfigValueFactory;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
 import java.time.LocalDateTime;
 
 import org.mockito.MockedStatic;
@@ -95,11 +94,7 @@ public class TDengineSinkWriterTest {
     }
 
     @Test
-    void testConvertDataType_withNull() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
-        // Use reflection to access the private method
-        Method method = TDengineSinkWriter.class.getDeclaredMethod("convertDataType", Object[].class);
-        method.setAccessible(true);
-
+    void testConvertDataTypeWithNull(){
         // Prepare test data
         LocalDateTime dateTime = LocalDateTime.of(2023, 4, 14, 15, 30, 45); // 2023-04-14 15:30:45
         Object[] input = {
@@ -119,21 +114,20 @@ public class TDengineSinkWriterTest {
                 45.67                           // Double remains unchanged
         };
 
-        // Invoke the private method
-        Object[] result = (Object[]) method.invoke(writer, (Object) input);
+        Object[] result = writer.convertDataType(input);
         // Verify the results
         assertArrayEquals(expectedOutput, result);
 
         // Test for an empty array
         Object[] input1 = {};
         Object[] expectedOutput1 = {};
-        Object[] result1 = (Object[]) method.invoke(writer, (Object) input1);
+        Object[] result1 = writer.convertDataType(input1);
         assertArrayEquals(expectedOutput1, result1, "Empty input array should return an empty output array.");
 
         // Test for an array containing only null
         Object[] input2 = {null};
         Object[] expectedOutput2 = {null};
-        Object[] result2 = (Object[]) method.invoke(writer, (Object) input2);
+        Object[] result2 = writer.convertDataType(input2);
         assertArrayEquals(expectedOutput2, result2, "Array with only null should return an array with null.");
     }
 }

--- a/seatunnel-connectors-v2/connector-tdengine/src/test/java/org/apache/seatunnel/connectors/seatunnel/tdengine/sink/TDengineSinkWriterTest.java
+++ b/seatunnel-connectors-v2/connector-tdengine/src/test/java/org/apache/seatunnel/connectors/seatunnel/tdengine/sink/TDengineSinkWriterTest.java
@@ -18,34 +18,32 @@
 
 package org.apache.seatunnel.connectors.seatunnel.tdengine.sink;
 
-
-import lombok.SneakyThrows;
-import org.apache.seatunnel.api.table.type.BasicType;
-import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
-import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
 import org.apache.seatunnel.shade.com.typesafe.config.Config;
 import org.apache.seatunnel.shade.com.typesafe.config.ConfigFactory;
 import org.apache.seatunnel.shade.com.typesafe.config.ConfigValueFactory;
+
+import org.apache.seatunnel.api.table.type.BasicType;
+import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import java.lang.reflect.InvocationTargetException;
-import java.time.LocalDateTime;
-
 import org.mockito.MockedStatic;
 import org.mockito.Mockito;
+
+import lombok.SneakyThrows;
+
 import java.sql.Connection;
 import java.sql.DriverManager;
 import java.sql.ResultSet;
 import java.sql.Statement;
+import java.time.LocalDateTime;
 import java.util.TimeZone;
 
-import static org.junit.jupiter.api.Assertions.*;
-import static org.mockito.ArgumentMatchers.anyString;
-import static org.mockito.Mockito.mockStatic;
-import static org.mockito.Mockito.when;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
 
-public class TDengineSinkWriterTest {
-    private TDengineSinkWriter writer;
+class TDengineSinkWriterTest {
+    TDengineSinkWriter writer;
 
     @SneakyThrows
     @BeforeEach
@@ -53,18 +51,19 @@ public class TDengineSinkWriterTest {
         SeaTunnelRowType rowType;
         Config config;
         TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
-        String[] fieldNames = new String[]{"id", "name", "description", "weight"};
+        String[] fieldNames = new String[] {"id", "name", "description", "weight"};
         SeaTunnelDataType<?>[] dataTypes =
-                new SeaTunnelDataType[]{
-                        BasicType.LONG_TYPE,
-                        BasicType.STRING_TYPE,
-                        BasicType.STRING_TYPE,
-                        BasicType.STRING_TYPE,
+                new SeaTunnelDataType[] {
+                    BasicType.LONG_TYPE,
+                    BasicType.STRING_TYPE,
+                    BasicType.STRING_TYPE,
+                    BasicType.STRING_TYPE,
                 };
         rowType = new SeaTunnelRowType(fieldNames, dataTypes);
         config =
                 ConfigFactory.empty()
-                        .withValue("url", ConfigValueFactory.fromAnyRef("jdbc:TAOS://localhost:6030/"))
+                        .withValue(
+                                "url", ConfigValueFactory.fromAnyRef("jdbc:TAOS://localhost:6030/"))
                         .withValue("database", ConfigValueFactory.fromAnyRef("test_db"))
                         .withValue("stable", ConfigValueFactory.fromAnyRef("test_stable"))
                         .withValue("username", ConfigValueFactory.fromAnyRef("root"))
@@ -77,41 +76,44 @@ public class TDengineSinkWriterTest {
         ResultSet mockResultSet = Mockito.mock(ResultSet.class);
 
         // Mock ResultSet behavior
-        when(mockResultSet.next()).thenReturn(true, false); // First call returns true, second call returns false
-        when(mockResultSet.getString("note")).thenReturn("TAG");
+        Mockito.when(mockResultSet.next())
+                .thenReturn(true, false); // First call returns true, second call returns false
+        Mockito.when(mockResultSet.getString("note")).thenReturn("TAG");
 
         // Mock Statement behavior
-        when(mockStatement.executeQuery("desc test_db.test_stable")).thenReturn(mockResultSet);
+        Mockito.when(mockStatement.executeQuery("desc test_db.test_stable"))
+                .thenReturn(mockResultSet);
 
         // Mock Connection behavior
-        when(mockConnection.createStatement()).thenReturn(mockStatement);
+        Mockito.when(mockConnection.createStatement()).thenReturn(mockStatement);
 
-        // Mock DriverManager behavior
-        try (MockedStatic<DriverManager> mockedStatic = mockStatic(DriverManager.class)) {
-            mockedStatic.when(() -> DriverManager.getConnection(anyString())).thenReturn(mockConnection);
+        try (MockedStatic<DriverManager> mockedStatic = Mockito.mockStatic(DriverManager.class)) {
+            Mockito.when(DriverManager.getConnection(Mockito.anyString()))
+                    .thenReturn(mockConnection);
             writer = new TDengineSinkWriter(config, rowType);
         }
     }
 
     @Test
-    void testConvertDataTypeWithNull(){
+    void testConvertDataTypeWithNull() {
         // Prepare test data
         LocalDateTime dateTime = LocalDateTime.of(2023, 4, 14, 15, 30, 45); // 2023-04-14 15:30:45
         Object[] input = {
-                null,                          // Test for null value
-                dateTime,                      // Test for LocalDateTime
-                "test_string",                 // Test for String
-                123,                           // Test for other types (Integer)
-                45.67                          // Test for other types (Double)
+            null, // Test for null value
+            dateTime, // Test for LocalDateTime
+            "test_string", // Test for String
+            123, // Test for other types (Integer)
+            45.67 // Test for other types (Double)
         };
 
         // Expected output
         Object[] expectedOutput = {
-                null,                           // null remains unchanged
-                "'2023-04-14 15:30:45.000'",    // LocalDateTime is converted to a formatted string with the specified timezone
-                "'test_string'",                // String is wrapped in single quotes
-                123,                            // Integer remains unchanged
-                45.67                           // Double remains unchanged
+            null, // null remains unchanged
+            "'2023-04-14 15:30:45.000'", // LocalDateTime is converted to a formatted string with
+            // the specified timezone
+            "'test_string'", // String is wrapped in single quotes
+            123, // Integer remains unchanged
+            45.67 // Double remains unchanged
         };
 
         Object[] result = writer.convertDataType(input);
@@ -122,12 +124,14 @@ public class TDengineSinkWriterTest {
         Object[] input1 = {};
         Object[] expectedOutput1 = {};
         Object[] result1 = writer.convertDataType(input1);
-        assertArrayEquals(expectedOutput1, result1, "Empty input array should return an empty output array.");
+        assertArrayEquals(
+                expectedOutput1, result1, "Empty input array should return an empty output array.");
 
         // Test for an array containing only null
         Object[] input2 = {null};
         Object[] expectedOutput2 = {null};
         Object[] result2 = writer.convertDataType(input2);
-        assertArrayEquals(expectedOutput2, result2, "Array with only null should return an array with null.");
+        assertArrayEquals(
+                expectedOutput2, result2, "Array with only null should return an array with null.");
     }
 }

--- a/seatunnel-connectors-v2/connector-tdengine/src/test/java/org/apache/seatunnel/connectors/seatunnel/tdengine/sink/TDengineSinkWriterTest.java
+++ b/seatunnel-connectors-v2/connector-tdengine/src/test/java/org/apache/seatunnel/connectors/seatunnel/tdengine/sink/TDengineSinkWriterTest.java
@@ -1,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.tdengine.sink;
+
+
+import lombok.SneakyThrows;
+import org.apache.seatunnel.api.table.type.BasicType;
+import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+import org.apache.seatunnel.shade.com.typesafe.config.ConfigFactory;
+import org.apache.seatunnel.shade.com.typesafe.config.ConfigValueFactory;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.time.LocalDateTime;
+
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.TimeZone;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+
+public class TDengineSinkWriterTest {
+    private TDengineSinkWriter writer;
+
+    @SneakyThrows
+    @BeforeEach
+    public void setup() {
+        SeaTunnelRowType rowType;
+        Config config;
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+        String[] fieldNames = new String[]{"id", "name", "description", "weight"};
+        SeaTunnelDataType<?>[] dataTypes =
+                new SeaTunnelDataType[]{
+                        BasicType.LONG_TYPE,
+                        BasicType.STRING_TYPE,
+                        BasicType.STRING_TYPE,
+                        BasicType.STRING_TYPE,
+                };
+        rowType = new SeaTunnelRowType(fieldNames, dataTypes);
+        config =
+                ConfigFactory.empty()
+                        .withValue("url", ConfigValueFactory.fromAnyRef("jdbc:TAOS://localhost:6030/"))
+                        .withValue("database", ConfigValueFactory.fromAnyRef("test_db"))
+                        .withValue("stable", ConfigValueFactory.fromAnyRef("test_stable"))
+                        .withValue("username", ConfigValueFactory.fromAnyRef("root"))
+                        .withValue("password", ConfigValueFactory.fromAnyRef("taosdata"))
+                        .withValue("timezone", ConfigValueFactory.fromAnyRef("UTC"));
+
+        // Mock JDBC objects
+        Connection mockConnection = Mockito.mock(Connection.class);
+        Statement mockStatement = Mockito.mock(Statement.class);
+        ResultSet mockResultSet = Mockito.mock(ResultSet.class);
+
+        // Mock ResultSet behavior
+        when(mockResultSet.next()).thenReturn(true, false); // First call returns true, second call returns false
+        when(mockResultSet.getString("note")).thenReturn("TAG");
+
+        // Mock Statement behavior
+        when(mockStatement.executeQuery("desc test_db.test_stable")).thenReturn(mockResultSet);
+
+        // Mock Connection behavior
+        when(mockConnection.createStatement()).thenReturn(mockStatement);
+
+        // Mock DriverManager behavior
+        try (MockedStatic<DriverManager> mockedStatic = mockStatic(DriverManager.class)) {
+            mockedStatic.when(() -> DriverManager.getConnection(anyString())).thenReturn(mockConnection);
+            writer = new TDengineSinkWriter(config, rowType);
+        }
+    }
+
+    @Test
+    void testConvertDataType_withNull() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        // Use reflection to access the private method
+        Method method = TDengineSinkWriter.class.getDeclaredMethod("convertDataType", Object[].class);
+        method.setAccessible(true);
+
+        // Prepare test data
+        LocalDateTime dateTime = LocalDateTime.of(2023, 4, 14, 15, 30, 45); // 2023-04-14 15:30:45
+        Object[] input = {
+                null,                          // Test for null value
+                dateTime,                      // Test for LocalDateTime
+                "test_string",                 // Test for String
+                123,                           // Test for other types (Integer)
+                45.67                          // Test for other types (Double)
+        };
+
+        // Expected output
+        Object[] expectedOutput = {
+                null,                           // null remains unchanged
+                "'2023-04-14 15:30:45.000'",    // LocalDateTime is converted to a formatted string with the specified timezone
+                "'test_string'",                // String is wrapped in single quotes
+                123,                            // Integer remains unchanged
+                45.67                           // Double remains unchanged
+        };
+
+        // Invoke the private method
+        Object[] result = (Object[]) method.invoke(writer, (Object) input);
+        // Verify the results
+        assertArrayEquals(expectedOutput, result);
+
+        // Test for an empty array
+        Object[] input1 = {};
+        Object[] expectedOutput1 = {};
+        Object[] result1 = (Object[]) method.invoke(writer, (Object) input1);
+        assertArrayEquals(expectedOutput1, result1, "Empty input array should return an empty output array.");
+
+        // Test for an array containing only null
+        Object[] input2 = {null};
+        Object[] expectedOutput2 = {null};
+        Object[] result2 = (Object[]) method.invoke(writer, (Object) input2);
+        assertArrayEquals(expectedOutput2, result2, "Array with only null should return an array with null.");
+    }
+}

